internal/pkg/agent/install/install.go

```diff
@@ -123,110 +176,69 @@ func Install(cfgFile, topPath string, unprivileged bool, log *logp.Logger, pt *p
 	}
 
 	pt.Describe("Copying install files")
-	err = copy.Copy(dir, topPath, copy.Options{
+	err = copy.Copy(sourceDir, topPath, copy.Options{
 		OnSymlink: func(_ string) copy.SymlinkAction {
 			return copy.Shallow
 		},
 		Sync:         true,
 		NumOfWorkers: int64(copyConcurrency),
 	})
+
 	if err != nil {
-		pt.Describe("Error copying files")
-		return utils.FileOwner{}, errors.New(
-			err,
-			fmt.Sprintf("failed to copy source directory (%s) to destination (%s)", dir, topPath),
-			errors.M("source", dir), errors.M("destination", topPath),
-		)
+		return errors.New(err, fmt.Sprintf("failed to copy source directory (%s) to destination (%s)", sourceDir, topPath), errors.M("source", sourceDir), errors.M("destination", topPath))
 	}
 	pt.Describe("Successfully copied files")
 
-	// place shell wrapper, if present on platform
-	if paths.ShellWrapperPath != "" {
-		pathDir := filepath.Dir(paths.ShellWrapperPath)
-		err = os.MkdirAll(pathDir, 0755)
-		if err != nil {
-			return utils.FileOwner{}, errors.New(
-				err,
-				fmt.Sprintf("failed to create directory (%s) for shell wrapper (%s)", pathDir, paths.ShellWrapperPath),
-				errors.M("directory", pathDir))
-		}
-		// Install symlink for darwin instead of the wrapper script.
-		// Elastic-agent should be first process that launchd starts in order to be able to grant
-		// the Full-Disk Access (FDA) to the agent and it's child processes.
-		// This is specifically important for osquery FDA permissions at the moment.
-		if runtime.GOOS == darwin {
-			// Check if previous shell wrapper or symlink exists and remove it so it can be overwritten
-			if _, err := os.Lstat(paths.ShellWrapperPath); err == nil {
-				if err := os.Remove(paths.ShellWrapperPath); err != nil {
-					return utils.FileOwner{}, errors.New(
-						err,
-						fmt.Sprintf("failed to remove (%s)", paths.ShellWrapperPath),
-						errors.M("destination", paths.ShellWrapperPath))
-				}
-			}
-			err = os.Symlink(filepath.Join(topPath, paths.BinaryName), paths.ShellWrapperPath)
-			if err != nil {
-				return utils.FileOwner{}, errors.New(
-					err,
-					fmt.Sprintf("failed to create elastic-agent symlink (%s)", paths.ShellWrapperPath),
-					errors.M("destination", paths.ShellWrapperPath))
-			}
-		} else {
-			// We use strings.Replace instead of fmt.Sprintf here because, with the
-			// latter, govet throws a false positive error here: "fmt.Sprintf call has
-			// arguments but no formatting directives".
-			shellWrapper := strings.Replace(paths.ShellWrapper, "%s", topPath, -1)
-			err = os.WriteFile(paths.ShellWrapperPath, []byte(shellWrapper), 0755)
-			if err != nil {
-				return utils.FileOwner{}, errors.New(
-					err,
-					fmt.Sprintf("failed to write shell wrapper (%s)", paths.ShellWrapperPath),
-					errors.M("destination", paths.ShellWrapperPath))
-			}
-		}
-	}
-
 	// create the install marker
 	if err := CreateInstallMarker(topPath, ownership); err != nil {
```

Comment:
I was expecting this call to be moved before the copy of other files, aren't we going to have the same issue if the copy of files above fails or gets interrupted for whatever reason ?

Reply:
So, the install marker is just there so `elastic-agent` doesn't complain during an uninstall operation. However, if that initial copy doesn't complete, we might not even have an `elastic-agent` binary or any files, and anything that is copied can either be removed via `rm` or just overwritten on  the next install. This is a less troublesome failure mode than creating the install marker at the end, where we'd have to remove service files, systemd config, etc, by hand.